emulator/blockchain.go

```diff
@@ -1182,6 +1208,34 @@ func (b *Blockchain) executeBlock() ([]*types.TransactionResult, error) {
 		}
 	}
 
+	script := templates.ReplaceAddresses(
+		systemChunkTransactionTemplate,
+		templates.Environment{
+			RandomBeaconHistoryAddress: b.GetChain().ServiceAddress().Hex(),
+		},
+	)
+	serviceKey := b.ServiceKey()
+	serviceAccount, _ := b.getAccount(flowgo.Address(b.serviceKey.Address))
+	tx := flowsdk.NewTransaction().
+		SetScript([]byte(script)).
+		SetGasLimit(9999).
+		AddAuthorizer(serviceKey.Address).
+		SetProposalKey(serviceKey.Address, 1, serviceAccount.Keys[1].SeqNumber).
+		SetPayer(serviceKey.Address).
+		SetReferenceBlockID(flowsdk.Identifier(b.pendingBlock.parentID))
+
+	signer, err := serviceKey.Signer()
+	if err != nil {
+		return nil, err
+	}
+
+	err = tx.SignEnvelope(serviceKey.Address, 1, signer)
+	if err != nil {
+		return nil, err
+	}
+	txn := convert.SDKTransactionToFlow(*tx)
+	b.pendingBlock.AddTransaction(*txn)
```

Comment:
I would take a note of which transaction index this is. then later in this function when executing the transaction with this index do `ctx :=  fvm.NewContextFromParent(blockContext,  fvm.WithRandomSourceHistoryCallAllowed(true))`

Reply:
Good catch :100: 
I forgot to enable this just for the system chunk transaction.
Updated in https://github.com/onflow/flow-emulator/pull/511/commits/2ee07522ad27fc650a07df3b7748b769493ad517
I was able to make do without using a transaction index for this :pray: